### PR TITLE
Enforce R-package dependencies in build_r.R before compilation begins

### PR DESCRIPTION
**Pull request by zkurtz**
I found myself forgetting to install a couple of R-package dependencies repeatedly on successive LightGBM installations. This wasted some time because compilation runs for quite a while before getting to the part where these packages are required.

### CODE REVIEW

File: build_r.R [hunk 1]
```diff
@@ -5,6 +5,22 @@
 # Sys.setenv("CXX" = "/usr/local/bin/g++-8")
 # Sys.setenv("CC" = "/usr/local/bin/gcc-8")
 
+# Identify R-packages required for build_r.R to complete
+pkgs = list(
+    installed = rownames(installed.packages()),
+    required = c("testthat", "roxygen2", "devtools")
+)
+pkgs$missing = setdiff(pkgs$required, pkgs$installed)
+if(length(pkgs$missing) > 0){
```
Comment (Laurae2) [inline build_r.R]: add spacing

File: build_r.R [hunk 2]
```diff
@@ -5,6 +5,22 @@
...
+    pkgs_string = paste0("    ", pkgs$missing, "\n")
+    message = paste0(
+        "The following R-packages are required for ",
+        "the LightGBM R-package install: \n",
+        paste0(pkgs_string, collapse = ''),
```
Comment (Laurae2) [inline build_r.R]: "" instead of '' for consistency

File: build_r.R [hunk 3]
```diff
@@ -5,6 +5,22 @@
 # Sys.setenv("CXX" = "/usr/local/bin/g++-8")
 # Sys.setenv("CC" = "/usr/local/bin/gcc-8")
 
+# Identify R-packages required for build_r.R to complete
```
Comment (jameslamb) [inline build_r.R]: Thanks @zkurtz , and apologies for the inconvenience.

But this is excessive complexity. Can you just add the following to the top of the script:

```
library(testthat)
library(roxygen2)
library(devtools)
```

Then anyone who doesn't have those will have this script immediately fail, and can just install them with `install.packages`

Comment (jameslamb) [reply build_r.R]: also please keep an eye on #629 ! We are trying to get CRAN-ready as fast as possible so hopefully you soon won't need to build from source.

**Comment by jameslamb** [comment]
@zkurtz looks good to me! Thanks for the change

@Laurae2 I will merge when all the CI stuff builds

**Comment by Laurae2** [comment]
@jameslamb To merge?

**Comment by jameslamb** [comment]
@Laurae2 yep looks good to me! I'll merge

**Pull request merged**